File: back/src/main/java/br/eng/eliseu/presente/model/filter/EventoReportFilter.java
package br.eng.eliseu.presente.model.filter;

import br.eng.eliseu.presente.model.core.BaseReportFilter;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class EventoReportFilter extends BaseReportFilter {

    private Long id;

    private Long clienteId;
    private Long eventoId;

    @Setter(AccessLevel.NONE)
    private Integer jaEscolheu; // null = todos, true = sim, false = não


    public void setJaEscolheu(Integer jaEscolheu) {
        if (jaEscolheu == null) jaEscolheu = -1;
        this.jaEscolheu = jaEscolheu;
    }

    public Integer getJaEscolheu() {
        return jaEscolheu;
    }

}

File: front/src/app/shared/model/filter/evento-report-filter.ts
import {BaseReportFilter} from "@/shared/model/core/base-report-filter";

export class EventoReportFilter extends BaseReportFilter{

    id?: number;

    clienteId?: number;
    eventoId?: number;
    jaEscolheu?: number; // null|-1 = todos, 1 = sim, 0 = não
}
